app/controllers/api_controller/reports.rb

```diff
@@ -44,5 +53,38 @@ def import_resource_reports(_type, _id, data)
       result, meta = MiqReport.import_from_hash(data["report"], options)
       action_result(meta[:level] == :info, meta[:message], :result => result)
     end
+
+    def schedule_resource_reports(_type, id, data)
+      schedule_data = fetch_schedule_data data
```

Comment:
do a resource_search to get the MiqReport first. i.e.

report = resource_search(id, type, collection_class(type))
report.add_schedule fetch_schedule_data(data)